programs/loft.sc

```diff
@@ -0,0 +1,356 @@
+__config() -> {
+    'stay_loaded'->true,
+    'commands'->{
+        'wand'->['wand'],
+        'paste <Block> <replace>'->['paste'],
+        'paste <Block>'->['paste',null],
+        'paste'->['paste','diamond_block',null],
+        'undo'->['undo']
+    },
+    'arguments'->{
+        'Block'->{'type'->'term', 'suggest'->[
+			'white_concrete','yellow_concrete','orange_concrete','purple_concrete',
+			'lime_concrete','red_concrete','green_concrete','magenta_concrete',
+			'brown_concrete','gray_concrete','light_gray_concrete','blue_concrete',
+			'light_blue_concrete','pink_concrete','cyan_concrete','black_concrete'
+			]
+		},
+        'replace'->{'type'->'term','suggest'->['air','water','lava']}
+    }
+};
+
+global_tool='phantom_membrane';
+global_block='diamond_block';
+global_replace=null;
+global_sets=[]; //array containing current information about the m splines created
+global_points=[]; //vector of maximum n points of the current spline that is being created by the player
+global_cset=[]; //grid of nxm points created based on global_sets by the function _normalise_set()
+global_a=[]; //grid of a coefficients needed for bezier surfaces
+global_b=[]; //grid of b coefficients needed for bezier surfaces
+global_c=[]; //grid of c coefficients needed for bezier surfaces
+global_affected_block=[]; //vector containing information about block that the script changed in the last paste operation
+global_history=[]; //vector containing information about block that the script changed in the last N paste operation
+
+wand()->(
+    print('the wand is:');
+    print(global_tool);
+);
+
+__on_player_breaks_block(player, block)->( //starts a new spline
+    item_tuple = query(player,'holds',hand='mainhand'):0;
+    if(item_tuple==global_tool,
+        schedule(0, _(outer(block)) -> set(pos(block), block));
+        _draw_surf(1);
+        global_points=[];
+        global_points+=pos(block);
+        global_sets:(length(global_sets))=global_points;
+        _normalise_set();
+        _draw_surf(10000)
```

Comment:
I understand the idea of just drawing for a long time, but the proper way to do this is to call a function that will then schedule itself. Something like 
```
_draw_surf_tick() -> (
  _draw_surf(12);
  if(global_draw_surf, schedule(10, '_draw_surf_tick') )
);
```
where here 10 is how often the surf is updated and 12 is a number just a bit over 10 so you don't get flickering in slower mashines. Both numbers acna of course be global values so thye are easily set.